src/remoteConnector.ts

```diff
@@ -771,62 +791,92 @@ export default class RemoteConnector extends Disposable {
 		}
 	}
 
-	public async checkRemoteConnectionSuccessful() {
-		const isRemoteExtensionHostRunning = async () => {
-			try {
-				// Invoke command from gitpot-remote extension to test if connection is successful
-				await vscode.commands.executeCommand('__gitpod.getGitpodRemoteLogsUri');
-				return true;
-			} catch {
-				return false;
-			}
-		};
+	private startHeartBeat(accessToken: string, connectionInfo: SSHConnectionParams) {
+		if (this.heartbeatManager) {
+			return;
+		}
 
-		const parseSSHDest = (sshDestStr: string): { user: string; hostName: string } | string => {
-			let decoded;
-			try {
-				decoded = JSON.parse(Buffer.from(sshDestStr, 'hex').toString('utf8'));
-			} catch {
-				// no-op
-			}
-			return decoded && typeof decoded.hostName === 'string' ? decoded : sshDestStr;
-		};
+		this.heartbeatManager = new HeartbeatManager(connectionInfo.gitpodHost, connectionInfo.workspaceId, connectionInfo.instanceId, accessToken, this.logger, this.telemetry);
+		this.logger.trace(`Heartbeat manager for workspace ${connectionInfo.workspaceId} (${connectionInfo.instanceId}) started`);
```

Comment:
logging host makes sense that if someone shares with us we know whether it is related to gitpod.io